docs/draft_production_docs/oadp-self-service.md

```diff
@@ -0,0 +1,484 @@
+
+To enable the approval workflow the DPA spec must be set as follows:
+```
+  nonAdmin:
+    enable: true
+    requireApprovalForBSL: true
+```
+Cluster administrators can view the NABSLApprovalRequest object in the openshift-adp namespace.
+
+```
+oc -n openshift-adp get NonAdminBackupStorageLocationRequests
+```
+Approval or rejection is accomplished by updating the NABSLApprovalRequest object.
+
+```
+spec:
+  approvalDecision: reject  [accept, reject]
+```
+
+If approved both the NABSL and the BSL are created.  The NABSL is created in the users namespace, while the BSL is created in the openshift-adp or default namespace.
+
+### User Creation of NABSL:
+
+A non-admin user can create a NABSL in their namespace.
+
+```
+apiVersion: oadp.openshift.io/v1alpha1
+kind: NonAdminBackupStorageLocation
+metadata:
+  name: nacuser1-nabsl
+  namespace: nacuser1
+spec: 
+  backupStorageLocationSpec:
+    config:
+    checksumAlgorithm: ""
+    profile: default
+    region: us-west-2
+    credential:
+      key: cloud
+      name: cloud-credentials
+    objectStorage:
+      bucket: bucket1uswest2
+      prefix: velero
+    provider: aws  
+```
+
+If the cluster administrator has enabled the requireApprovalForBSL flag then the NABSL will be in the Pending state until the cluster administrator approves the NABSL.
+
+
+
+### NAB / NAR Status
+
+#### Phase
+The phase field is a simple one high-level summary of the lifecycle of the objects, that only moves forward. Once a phase changes, it can not return to the previous value.
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| New | *NonAdminBackup/NonAdminRestore* resource was accepted by the NAB/NAR Controller, but it has not yet been validated by the NAB/NAR Controller |
+| BackingOff | *NonAdminBackup/NonAdminRestore* resource was invalidated by the NAB/NAR Controller, due to invalid Spec. NAB/NAR Controller will not reconcile the object further, until user updates it |
+| Created | *NonAdminBackup/NonAdminRestore* resource was validated by the NAB/NAR Controller and Velero *Backup/restore* was created. The Phase will not have additional information about the *Backup/Restore* run |
+| Deletion | *NonAdminBackup/NonAdminRestore* resource has been marked for deletion. The NAB/NAR Controller will delete the corresponding Velero *Backup/Restore* if it exists. Once this deletion completes, the *NonAdminBackup/NonAdminRestore* object itself will also be removed |
+
+
+
+
+
+## Advanced Cluster Administrator Features
+
+### Cluster Administrator Enforceable Spec Fields
+There are several types of cluster scoped objects that non-admin users should not have access to backup or restore.  OADP self-service automatically excludes the following list of cluster scoped resources from being backed up or restored.
+
+* SCCs
+* ClusterRoles
+* ClusterRoleBindings
+* CRDs
+* PriorityClasses
+* virtualmachineclusterinstancetypes
+* virtualmachineclusterpreferences
+
```

Comment:
Should we add a note that these cover only the velero core and OADP default plugins ? We are not responsible if they have a custom plugin in place, right ?

Reply:
custom plugins are not supported period.